backend/src/main/java/com/tourneynizer/tourneynizer/model/Match.java

```diff
@@ -49,28 +40,11 @@ public void setTournamentId(long tournament_id) {
         this.tournament_id = tournament_id;
     }
 
-    public void setteam1Id(Long team1Id) {
-        if (this.team2Id != null && this.team2Id.equals(team1Id)) {
-            throw new IllegalArgumentException("Teams cannot play themselves");
-        }
-        if (this.refId != null && this.refId.equals(team1Id)) {
-            throw new IllegalArgumentException("Teams cannot ref their own game");
-        }
-        this.team1Id = team1Id;
-    }
-
-    public void setteam2Id(Long team2Id) {
-        if (this.team1Id != null && this.team1Id.equals(team2Id)) {
-            throw new IllegalArgumentException("Teams cannot play themselves");
-        }
-        if (this.refId != null && this.refId.equals(team2Id)) {
-            throw new IllegalArgumentException("Teams cannot ref their own game");
-        }
-        this.team2Id = team2Id;
-    }
 
     public void setRefId(Long refId) {
-        if (refId != null && (Objects.equals(refId, team1Id) || Objects.equals(refId, team2Id))) {
+        long firstChild = matchChildren.first();
+        long secondChild = matchChildren.second();
+        if (refId != null && (Objects.equals(refId, firstChild) || Objects.equals(refId, secondChild))) {
```

Comment:
Here refId is a team ID and firstChild/secondChild are either match IDs or team IDs depending on their position. Update the first and second methods to make sure they return only the winner of the first and second match, not the match itself.